mon-pix/app/authenticators/oidc.js

```diff
@@ -0,0 +1,62 @@
+import OIDCAuthenticator from 'ember-simple-auth-oidc/authenticators/oidc';
+import getAbsoluteUrl from 'ember-simple-auth-oidc/utils/absoluteUrl';
+
+import config from 'ember-simple-auth-oidc/config';
+
+const {
+  host,
+  clientId,
+  afterLogoutUri,
+  endSessionEndpoint,
+} = config;
+
+export default OIDCAuthenticator.extend({
+
+  async authenticate({ code, redirectUri }) {
+    const bodyObject = {
+      code,
+      client_id: clientId,
+      redirect_uri: redirectUri,
+    };
+
+    const body = Object.keys(bodyObject)
+      .map((k) => `${k}=${encodeURIComponent(bodyObject[k])}`)
+      .join('&');
+
+    const response = await fetch('/api/token/pole-emploi', {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      body,
+    });
+
+    const data = await response.json();
+
+    return {
+      access_token: data.access_token,
+      id_token: data.id_token,
+      redirectUri,
+    };
+  },
+
+  singleLogout(idToken) {
```

Comment:
Pourquoi override la fonction de la lib en mettant le même code.

D'ailleurs par rapport au problème de déconnexion, cette fonction devrait bien dire au serveur d'autorisation PE d'invalider la session. (Voir la documentation)

 /**
   * Invalidates the current session (of this application) and calls the
   * `end-session` endpoint of the authorization server, which will
   * invalidate all sessions which are handled by the authorization server
   * (possible for multiple applications).
   *
   * @param {String} idToken The id_token of the session to invalidate
   */

En complément le lien d'@AndreiaPena qui parle de potentiel bug PE de coté https://www.mathieupassenaud.fr/logout-oauth2/

Reply:
La différence : 
Fonction d'origine : 
```javascript
...
if (afterLogoutUri) {
  params.push(`post_logout_redirect_uri=${getAbsoluteUrl(afterLogoutUri)}`);
}
...
```

Fonction modifiée : 
```javascript
...
if (afterLogoutUri) {
  params.push(`redirect_uri=${getAbsoluteUrl(afterLogoutUri)}`);
}
...
```